www/include/reporting/dashboard/csvExport/csv_ServiceGroupLogs.php

```diff
@@ -108,102 +129,158 @@
+    . _("Service") . ";"
+    . $stringOk . " %;"
+    . $stringOk . " " . $stringMeanTime . " %;"
+    . $stringOk . " " . $stringAlert . ";"
+    . $stringWarning . " %;"
+    . $stringWarning . " " . $stringMeanTime . " %;"
+    . $stringWarning . " " . $stringAlert . ";"
+    . $stringCritical . " %;"
+    . $stringCritical . " " . $stringMeanTime . " %;"
+    . $stringCritical . " " . $stringAlert . ";"
+    . $stringUnknown . " %;"
+    . $stringUnknown . $stringMeanTime . " %;"
+    . $stringUnknown . " " . $stringAlert . ";"
+    . $stringDowntime . " %;"
+    . $stringUndetermined . "\n";
 
 foreach ($stats as $key => $tab) {
     if ($key != "average") {
-        echo $tab["HOST_NAME"] . ";" . $tab["SERVICE_DESC"] . ";"
-            . $tab["OK_TP"] . "%;" . $tab["OK_MP"] . "%;" . $tab["OK_A"] . ";"
-            . $tab["WARNING_TP"] . "%;" . $tab["WARNING_MP"] . "%;" . $tab["WARNING_A"] . ";"
-            . $tab["CRITICAL_TP"] . "%;" . $tab["CRITICAL_MP"] . "%;" . $tab["CRITICAL_A"] . ";"
-            . $tab["UNKNOWN_TP"] . "%;" . $tab["UNKNOWN_MP"] . "%;" . $tab["UNKNOWN_A"] . ";"
-            . $tab["MAINTENANCE_TP"] . " %;" . $tab["UNDETERMINED_TP"]. "%\n";
+        echo $tab["HOST_NAME"] . ";"
+            . $tab["SERVICE_DESC"] . ";"
+            . $tab["OK_TP"] . "%;"
+            . $tab["OK_MP"] . "%;"
+            . $tab["OK_A"] . ";"
+            . $tab["WARNING_TP"] . "%;"
+            . $tab["WARNING_MP"] . "%;"
+            . $tab["WARNING_A"] . ";"
+            . $tab["CRITICAL_TP"] . "%;"
+            . $tab["CRITICAL_MP"] . "%;"
+            . $tab["CRITICAL_A"] . ";"
+            . $tab["UNKNOWN_TP"] . "%;"
+            . $tab["UNKNOWN_MP"] . "%;"
+            . $tab["UNKNOWN_A"] . ";"
+            . $tab["MAINTENANCE_TP"] . " %;"
+            . $tab["UNDETERMINED_TP"]. "%\n";
     }
 }
 echo "\n\n";
 
-/*
- * Services group stats evolution
- */
-echo _("Day") . ";" . _("Duration") . ";"
-    . $stringOk . " " . $stringMeanTime . ";" . $stringOk . " " . $stringAlert . ";"
-    . $stringWarning . " " . $stringMeanTime . ";" . $stringWarning . " " . $stringAlert . ";"
-    . $stringUnknown . " " . $stringMeanTime . ";" . $stringUnknown . " " . $stringAlert . ";"
-    . $stringCritical . " " . $stringMeanTime . ";" . $stringCritical . " " . $stringAlert . ";"
+// Services group stats evolution
+echo _("Day") . ";"
+    . _("Duration") . ";"
+    . $stringOk . " " . $stringMeanTime . ";"
+    . $stringOk . " " . $stringAlert . ";"
+    . $stringWarning . " " . $stringMeanTime . ";"
+    . $stringWarning . " " . $stringAlert . ";"
+    . $stringUnknown . " " . $stringMeanTime . ";"
+    . $stringUnknown . " " . $stringAlert . ";"
+    . $stringCritical . " " . $stringMeanTime . ";"
+    . $stringCritical . " " . $stringAlert . ";"
     . _("Day") . "\n";
 
+$dbResult = $pearDB->query(
+    "SELECT `service_service_id` FROM `servicegroup_relation` " .
+    "WHERE `servicegroup_sg_id` = '" . $servicegroupId . "'"
```

Comment:
In this case, using a prepare statement is hightly recommended

Reply:
Refacto the whole code is out of my scope...  ;)